### PR TITLE
Create API for posts and comments

### PR DESCRIPTION
**Pull request by rajanmaghera**
This PR uses the Django Rest Framework to create a simple API for posts and comments. 

It uses custom routers and serializers to match the format of the api that we want. This makes it very easy to extend in the future.

### CODE REVIEW

File: quickcomm/tests/test_api_authors.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        self.assertEqual(req.status_code, 405)
+
+    def test_get_author(self):
+        """Test that we can get an author"""
+        req = self.client.get('/api/authors/{}/'.format(self.author.id))
+        self.assertEqual(req.status_code, 200)
+        self.assertEqual(req.data['type'], 'author')
+        self.assertNotEquals(req.data['type'], 'authors')
+
+        self.assertEqual(req.data['id'], str(self.author.id))
```
Comment (Jdvdb) [inline quickcomm/tests/test_api_authors.py:56]: this line is failing when I run tests.
AssertionError: 'http://testserver/api/authors/7758be87-5e14-43dc-a814-97d477664b0c/' != '7758be87-5e14-43dc-a814-97d477664b0c'

**Comment by Jdvdb** [comment]
what link should I be using to query authors? i've tried http://127.0.0.1:8000/api/authors and a couple others but I either get a 301 or 404 when using curl.

**Comment by rajanmaghera** [comment]
> what link should I be using to query authors? i've tried http://127.0.0.1:8000/api/authors and a couple others but I either get a 301 or 404 when using curl.

The models might have changed. Have you tried migrating the models and installing all the new dependencies in the requirements.txt file?

**Comment by Jdvdb** [comment]
> The models might have changed. Have you tried migrating the models and installing all the new dependencies in the requirements.txt file?
yep, I had to install some more dependencies to get things running and migrated stuff too. Also went in and create an author just to make sure there was something to get. My curl requests come back empty and I see django say '"GET /api/authors HTTP/1.1" 301 0'

**Comment by rajanmaghera** [comment]
That's a bit weird, especially if any API tests pass.

You might have to clear your DB and/or browser cache, and try once more. Make sure to have the trailing slash in all of the requests. Also try both 127.0.0.1 and localhost.

**Comment by Jdvdb** [comment]
I think i was just missing the trailing slash and I was able to see a result. I don't have time right now to finish really reviewing the PR so if some other people could check quick that would be great!

**Pull request merged**